src/interfacers/EmonHubBleInterfacer.py

```
@@ -0,0 +1,168 @@
+        else:
+            c.nodeid = 1
+
+        return c
+
+    def set(self, **kwargs):
+
+        for key, setting in self._private_settings.iteritems():
+            # Decide which setting value to use
+            if key in kwargs.keys():
+                setting = kwargs[key]
+            else:
+                setting = self._private_settings[key]
+
+            # Ignore unchanged
+            if key in self._settings and self._settings[key] == setting:
+                continue
+            elif key == 'read_interval':
+                setting = float(setting)
+            else:
+                self._log.warning("'%s' is not valid for %s: %s" % (str(setting), self.name, key))
+                continue
+
+            self._log.debug('Setting {}: {}'.format(key, setting))
+            self._private_settings[key] = setting
+
+        # include kwargs from parent
+        super(EmonHubBleInterfacer, self).set(**kwargs)
+
+    def _get_temperature(self):
+        val = self._temperature.read()
+        (val,) = struct.unpack('h', val)
+        return val
+
+    def _get_humidity(self):
+        val = self._humidity.read()
+        (val,) = struct.unpack('h', val)
+        return val
+
+    def _get_bat_level(self):
+        val = self._bat_level.read()
+        (val,) = struct.unpack('B', val)
+
+        # The battery reading is very noisy - do a simple average
+        self._bat_readings.insert(0, val)
+        self._bat_readings = self._bat_readings[0:20]
+
+        val = sum(self._bat_readings)/float(len(self._bat_readings))
+        #self._log.debug('Batt: {} -> {}'.format(self._bat_readings, val))
+        
+        return round(val)
+
+    def _connect(self):
+        self._log.debug("Connecting to BLE address {}...".format(self._addr))
+        self._ble = None
+
+        try:
+            self._ble = btle.Peripheral(self._addr)
+        except btle.BTLEException as e:
+            self._log.error(e)
```

Use self._log.exception and give a useful message (not "e") to avoid swallowing the context of the exception.

---

Something like `self._log.exception("Failed to read from BTLE device")` would probably be fine

---

done